point prev
point next
float32 angle
float32[] ranges